uwblueprint/extend-a-family
# Forgot password

## Notion ticket link
[Forgot Password] https://www.notion.so/uwblueprintexecs/Forgot-Password-9caf66ce42134dd69b5a856f30c3f090


## Implementation description
* Create a new endpoint `POST /forgotPassword` in `authRoutes`. The request body takes a field `email`. Create and use a `forgotPasswordRequestValidator` in `authValidators` that checks if the provided email is a string.

In the endpoint:

* Check if the email address exists with `userService.getUserByEmail` (you’ll notice this function throws an error with message `"EMAIL_NOT_FOUND"` if the email doesn’t exist).
* If the email address doesn’t exist, return an error with body `{ error: "EMAIL_NOT_FOUND" }`
* If the email address does exist, send a reset password email with `authService.resetPassword`



## Steps to test
1. Open the endpoint in Postman (should be a POST requesting to the endpoint http://localhost:8080/auth/forgotPassword)
2. Try posting a variety of different emails in different requests in a format like the following:
      {
          "email": "email@email.com"
      }
3. If the email is registered (in the Firebase https://console.firebase.google.com/u/1/project/extendafamily-7613e/authentication/users), you should get a password update email in the inbox however, if it is not a registered email, the response should be "EMAIL_NOT_FOUND".


## What should reviewers focus on?
*  Make sure the right responses are going through and code clarity if you'd like
